### PR TITLE
[RR] fixed REST Exception handling when EJBException is thrown

### PR DESCRIPTION
**Pull request by michalovjan**
### Checklist:

* [ ] Have you added a note in the [CHANGELOG wiki](https://github.com/project-ncl/pnc/wiki/Changelog) for your change if user-facing?
* [ ] Have you added unit tests for your change?

@matedo1 This should resolve your message and status code issues.

### CODE REVIEW

File: rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    @Override public Response toResponse(EJBException exception) {
+        Throwable t = exception.getCause();
+        ExceptionMapper mapper = providers.getExceptionMapper(t.getClass());
+        log.info("Unwrapping " + t.getClass().getSimpleName() + " from EJBException and passing in it to its appropriate ExceptionMapper");
+
+        if( mapper != null ) {
+            return mapper.toResponse(t);
+        }
+        else {
+            log.error("Could not find exception mapper for exception " + t);
```
Comment (janinko) [inline rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java]: Should there be `t.getClass()`?

File: rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    @Override public Response toResponse(EJBException exception) {
+        Throwable t = exception.getCause();
+        ExceptionMapper mapper = providers.getExceptionMapper(t.getClass());
+        log.info("Unwrapping " + t.getClass().getSimpleName() + " from EJBException and passing in it to its appropriate ExceptionMapper");
+
+        if( mapper != null ) {
+            return mapper.toResponse(t);
+        }
+        else {
```
Comment (janinko) [inline rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java]: code style
`} else {`

File: rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    private static final Logger log = LoggerFactory.getLogger(EJBExceptionMapper.class);
+
+    @Context
+    private Providers providers;
+
+    @Override public Response toResponse(EJBException exception) {
+        Throwable t = exception.getCause();
+        ExceptionMapper mapper = providers.getExceptionMapper(t.getClass());
+        log.info("Unwrapping " + t.getClass().getSimpleName() + " from EJBException and passing in it to its appropriate ExceptionMapper");
```
Comment (janinko) [inline rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java]: I would change the level to debug.

File: rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java (hunk 1 above)
Comment (michalovjan) [reply rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java]: Yes, the wording is wrong there. Hmm, do we want to log just the Exception name or the whole stacktrace? This is a very specific case that should never happen so logging the whole stacktrace would be better for investigation I guess?

File: rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java (hunk 3 above)
Comment (michalovjan) [reply rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java]: Why not.

File: rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java (hunk 2 above)
Comment (michalovjan) [reply rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java]: My bad :D

File: rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java (hunk 1 above)
Comment (janinko) [reply rest-new/src/main/java/org/jboss/pnc/rest/provider/EJBExceptionMapper.java]: Yeah, you are right, we should log whole stacktrace.

**Pull request merged**